antidox/perspective.py

```diff
@@ -147,49 +152,60 @@ def main(argv):
   parser.add_argument('--csv_file', help='choose CSV file to process')
   parser.add_argument('--wiki_pagename', help='insert the talk page name')
   parser.add_argument('--content', help='specify a column in dataset to retreive data from')
+  parser.add_argument('--output', help='path for output file')
+  parser.add_argument('--suffix', help='output file suffix')
+  parser.add_argument('--project', help='project id for bigquery table')
+  parser.add_argument('--wiki_pagename', help='wikipedia page name')
   args = parser.parse_args(argv)
-
-  apikey_data, perspective, dlp = get_client()
-  pii_results = open("pii_results.txt", "w+")
-  toxicity_results = open("toxicity_results.txt", "w+")
-
-  if args.wiki_pagename:
-    wiki_response = get_wikipage(args.wiki_pagename)
-    wikitext = wiki_clean(wiki_response)
-    text = wikitext.split("\n")
-  if args.csv_file:
-    text = pd.read_csv(args.csv_file)
-  if args.sql_query:
-    big_q = bigquery.Client.from_service_account_json('querykey.json')
-    text = use_query(args.content, args.sql_query, big_q)
-
-  for line in text:
-    if not line:
-      continue
-    dlp_response = dlp_request(dlp, apikey_data, line)
-    try:
-      perspective_response = perspective_request(perspective, line)
-    # Perspective can't handle language errors at this time
-    except google_api_errors.HttpError as err:
-      print("Error:", err)
-    has_pii_bool, pii_type = contains_pii(dlp_response)
-    if has_pii_bool:
-      pii_results.write(str(line)+"\n"+'contains pii?'+"Yes"+"\n"
-                        +str(pii_type)+"\n"
-                        +"==============================================="+"\n")
-    if contains_toxicity(perspective_response):
-      toxicity_results.write(str(line)+"\n" +"contains TOXICITY?:"+
-                             "Yes"+"\n"+
-                             str(perspective_response['attributeScores']
-                                 ['TOXICITY']['summaryScore']['value'])+"\n"
-                             +"=========================================="+"\n")
-
-  toxicity_results.close()
-  pii_results.close()
-    # print('dlp result:', json.dumps(dlp_response, indent=2))
-    # print ("contains_toxicity:", json.dumps(perspective_response, indent=2))
-
-
+  apikey_data, perspective, dlp = get_client('api_key.json')
+  with beam.Pipeline(options=PipelineOptions()) as pipeline:
+    if args.wiki_pagename:
+      wiki_response = get_wikipage(args.wiki_pagename)
+      wikitext = wiki_clean(wiki_response)
+      comments = p | beam.create(wikitext.split("\n"))
+    if args.csv_file:
+      comments = pipeline | 'ReadMyFile' >> beam.io.ReadFromText(args.csv_file)
+    if args.sql_query:
```

Comment:
Can you give updated instructions in the README.md file for how to run this with the wikidetox table we have used for testing?  I tried this command line but got errors:
`python3 -m antidox.perspective  --output='output' --suffix='txt' --sql_query="SELECT cleaned_content FROM \`wikidetox-viz.wikiconv2_201904.en_cleaned_conv\` LIMIT 10"`

Is there something I need to run first to authenticate with bigquery?  It seems like we are no longer using `querykey.json`

Reply:
done

Reply:
have to use --content command line arg as well